crates/engine/tree/src/tree/mod.rs

```diff
@@ -421,25 +429,32 @@ where
         }
     }
 
+    /// Creates a new `EngineApiTreeHandlerImpl` instance and spawns it in its
+    /// own thread. Returns the receiver end of a `EngineApiEvent` unbounded
+    /// channel to receive events from the engine.
     #[allow(clippy::too_many_arguments)]
-    fn spawn_new(
+    pub fn spawn_new(
         provider: P,
         executor_provider: E,
         consensus: Arc<dyn Consensus>,
         payload_validator: ExecutionPayloadValidator,
         incoming: Receiver<FromEngine<BeaconEngineMessage<T>>>,
-        state: EngineApiTreeState,
```

Comment:
state and header are initialized in the constructor, is there any case when we want to inject them?

Reply:
I'm thinking we should have the header as an argument, or initialize it differently. For example we probably want to set it as the highest persisted block on node startup

Reply:
makes sense, done here https://github.com/paradigmxyz/reth/pull/9621/commits/7881e2e2f5801a70b539f519c47311e2c248573f